src/vstest.console/CommandLine/InferHelper.cs

```diff
@@ -25,11 +25,12 @@ internal InferHelper(IAssemblyMetadataProvider assemblyMetadataProvider)
         }
 
         /// <summary>
-        /// Determines Architecture from sources.
+        /// Determines Architecture from sources and returns true if source architectures are incompatible
         /// </summary>
-        public Architecture AutoDetectArchitecture(List<string> sources, IDictionary<string, Architecture> sourcePlatforms)
+        public bool TryGetCompatibleArchitecture(List<string> sources, IDictionary<string, Architecture> sourcePlatforms, out Architecture inferredArchitecture)
```

Comment:
This functions is still auto detecting Fx, please rename it accordingly.

Reply:
Renamed to `TryGetAutoDetectCompatibleArcitecture `and `TryGetAutoDetectCompatibleFramework`